### PR TITLE
Add xdp test cases and some improvements

### PR DESCRIPTION
**Pull request by squirrelsc**
See commits for details.

### CODE REVIEW

File: microsoft/testsuites/xdp/xdptools.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    @property
+    def can_install(self) -> bool:
+        ethtool = self.node.tools[Ethtool]
+        statistics = ethtool.get_device_statistics(self.node.nics.default_nic)
+
+        # check if xdp supported on nic
+        if not any("xdp_drop" in x for x in statistics):
+            raise UnsupportedDistroException(
+                self.node.os,
+                "Cannot find xdp_drop in ethotool statistics. "
```
Comment (LiliDeng) [inline microsoft/testsuites/xdp/xdptools.py]: ethtool

File: microsoft/testsuites/xdp/xdptools.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        super()._initialize(*args, **kwargs)
+        self._command: PurePath = PurePath(self._default_command)
+        self._gro_lro_settings: Dict[str, DeviceGroLroSettings] = {}
+
+    def _install(self) -> bool:
+        # install dependencies
+
+        config_envs: Dict[str, str] = {}
+        if isinstance(self.node.os, Debian):
+            if self.node.os.information.version < "18.4.0":
```
Comment (LiliDeng) [inline microsoft/testsuites/xdp/xdptools.py:82]: this works for ubuntu, how about debian images, current the latest debian image is 11

File: microsoft/testsuites/xdp/xdptools.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+            config_envs.update({"CLANG": "clang-10", "LLC": "llc-10"})
+
+        elif isinstance(self.node.os, Fedora):
+            self.node.os.install_packages(
+                "llvm-toolset elfutils-devel m4 wireshark perf make gcc tc "
+                # pcaplib
+                "http://rpmfind.net/linux/centos/8.5.2111/PowerTools/"
+                "x86_64/os/Packages/libpcap-devel-1.9.1-5.el8.x86_64.rpm"
+            )
```
Comment (LiliDeng) [inline microsoft/testsuites/xdp/xdptools.py:109]: seems it is only for centos 8.5 distro? if lower distro version used, will it fail?

File: microsoft/testsuites/xdp/xdptools.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            expected_exit_code=0,
+            expected_exit_code_failure_message="failed on configure xdp "
+            "tools before make.",
+        )
+        make = self.node.tools[Make]
+        make.make(
+            arguments="",
+            cwd=self._code_path,
+            update_envs={"ARCH": "x86_64"},
+            thread_count=1,
```
Comment (LiliDeng) [inline microsoft/testsuites/xdp/xdptools.py:139]: why use 1?

Comment (squirrelsc) [reply microsoft/testsuites/xdp/xdptools.py:139]: There are errors, if make with multi-threads. The program may not be ready for concurrent build, but our make tool use multi-thread by default.

File: microsoft/testsuites/xdp/xdptools.py (hunk 3 above)
Comment (squirrelsc) [reply microsoft/testsuites/xdp/xdptools.py:109]: The tool is not 1:1 mapping, it's hard to meet all OS versions. I tested it with Redhat 8.4, it works. The Redhat 8.3 doesn't support xdp. so it should be ok.

File: microsoft/testsuites/xdp/xdptools.py (hunk 4 above)
Comment (LiliDeng) [reply microsoft/testsuites/xdp/xdptools.py:139]: please add comments

**Comment by LiliDeng** [comment]
image
one typo in commit message.

File: microsoft/testsuites/xdp/xdptools.py (hunk 2 above)
Comment (squirrelsc) [reply microsoft/testsuites/xdp/xdptools.py:82]: this place is to compat with 1804. Lower version doesn't support xdp, higher version use the default toolchain. I haven't test with debian. Will fix debian problems after test with chaos monkey.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**